app/src/main/java/com/example/demomvp/data/source/CocktailRepository.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.demomvp.data.source
+
+import com.example.demomvp.data.model.Cocktail
+import com.example.demomvp.data.source.remote.CocktailRemoteDataSource
+import com.example.demomvp.data.source.remote.OnFetchDataJsonListener
+
+class CocktailRepository private constructor(
+    private val remote: CocktailDataSource.Remote
+) {
+
+    private object Holder {
+        val INSTANCE = CocktailRepository(
+            remote = CocktailRemoteDataSource.instance
+        )
+    }
+
+    fun getCocktail(listener: OnFetchDataJsonListener<MutableList<Cocktail>>) {
+        remote.getCocktails(listener)
+    }
+
+    companion object {
+        val instance: CocktailRepository by lazy { Holder.INSTANCE }
```

Comment:
companion object {
        private var instance: SearchRepository? = null
        fun getInstance() =
            instance ?: SearchRepository(RemoteDataSource.getInstance())
                .also { instance = it }
    }
sugest dùng singleton như thế này